parsec/backend/memory/user.py

```diff
@@ -285,26 +275,33 @@ async def find_humans(
             )
             for user in data
         ]
-
+        results = sorted(
+            [res for res in results if res.human_handle], key=lambda r: r.user_id.lower()
+        )
         if omit_revoked:
             results = [res for res in results if not res.revoked]
 
         # PostgreSQL does case insensitive sort
-        humans = sorted(
-            [res for res in results if res.human_handle],
-            key=lambda r: (r.human_handle.label.lower(), r.user_id.lower()),  # type: ignore
-        )
-        non_humans = sorted(
-            [res for res in results if not res.human_handle], key=lambda r: r.user_id.lower()
-        )
-
-        if omit_non_human:
+        if omit_non_human or query:
+            total = len(results)
+            results = results[(page - 1) * per_page : page * per_page]
+            humans = sorted(
+                [res for res in results if res.human_handle], key=lambda r: r.human_handle.label
+            )
             results = humans
         else:
-            # Keeping non-human last
+            total = len(results)
+            results = results[(page - 1) * per_page : page * per_page]
+            humans = sorted(
+                [res for res in results if res.human_handle], key=lambda r: r.human_handle.label
+            )
+            non_humans = [res for res in results if not res.human_handle]
             results = [*humans, *non_humans]
 
-        return (results[(page - 1) * per_page : page * per_page], len(results))
+        if not results:
```

Comment:
This shouldn't be needed given we can do `total = len(results)` even if results is an empty list

Reply:
done